### ISSUE TITLE
Enigmatic behavior of transaction_hash property in --lite-mode

### ISSUE THREAD

**Issue by mikiw** [Shard-Labs/starknet-devnet#247]
**Describe the bug**
Lite mode is not working as expected, ```transaction_hash``` should be 0x0 instead of 64 character hash.

**Not reproducible on alpha-goerli**

- [x] This issue is only present on Devnet and cannot be reproduced on alpha-goerli (check the box if true).

**To Reproduce**

1. Run ```poetry run starknet-devnet --lite-mode```

2. Run ```starknet deploy --contract "test/artifacts/contracts/cairo/contract.cairo/contract.json" --inputs "0" --no_wallet --gateway_url http://127.0.0.1:5050/ --feeder_gateway_url http://127.0.0.1:5050/```

Deploy transaction was sent.
Contract address: 0x00917c9a2fd86e08aadfcdf3f6212ad121d43c920454c501c6aa3e3963449980
Transaction hash: 0x0

3. Run ```starknet get_block --number 0 --no_wallet --gateway_url http://127.0.0.1:5050/ --feeder_gateway_url http://127.0.0.1:5050/``` - first block looks ok, but there is no transaction_hash.

```
{
    "block_hash": "0x0",
    "block_number": 0,
    "gas_price": "0x174876e800",
    "parent_block_hash": "0x0",
    "sequencer_address": "0x4bbfb0d1aab5bf33eec5ada3a1040c41ed902a1eeb38c78a753d6f6359f1666",
    "starknet_version": "0.9.1",
    "state_root": "0000000000000000000000000000000000000000000000000000000000000000",
    "status": "ACCEPTED_ON_L2",
    "timestamp": 1662020400,
    "transaction_receipts": [],
    "transactions": []
}
```

4. Run ```starknet get_block --number 1 --no_wallet --gateway_url http://127.0.0.1:5050/ --feeder_gateway_url http://127.0.0.1:5050/``` - next block has transaction_hash with 64 characters.

```
{
    "block_hash": "0x1",
    "block_number": 1,
    "gas_price": "0x174876e800",
    "parent_block_hash": "0x0",
    "sequencer_address": "0x4bbfb0d1aab5bf33eec5ada3a1040c41ed902a1eeb38c78a753d6f6359f1666",
    "starknet_version": "0.9.1",
    "state_root": "0000000000000000000000000000000000000000000000000000000000000000",
    "status": "ACCEPTED_ON_L2",
    "timestamp": 1662020408,
    "transaction_receipts": [
        {
            "actual_fee": "0x0",
            "events": [],
            "execution_resources": {
                "builtin_instance_counter": {
                    "bitwise_builtin": 0,
                    "ecdsa_builtin": 0,
                    "output_builtin": 0,
                    "pedersen_builtin": 0,
                    "range_check_builtin": 0
                },
                "n_memory_holes": 0,
                "n_steps": 40
            },
            "l2_to_l1_messages": [],
            "transaction_hash": "0x567ec53cc336686b56ee6d466dad00b9b866976d0cc05fb056ee36da8681b98",
            "transaction_index": 0
        }
    ],
    "transactions": [
        {
            "class_hash": "0x757a84aa38bf4ad191a7dfea2e8146fc7f3c4aa6090a8f0bddd7b688f0b24c",
            "constructor_calldata": [
                "0x0"
            ],
            "contract_address": "0x917c9a2fd86e08aadfcdf3f6212ad121d43c920454c501c6aa3e3963449980",
            "contract_address_salt": "0x1c1bb9aeeb944410a52c88aa3ad352dc19786ba03014d916d11b2ff86b915a8",
            "transaction_hash": "0x567ec53cc336686b56ee6d466dad00b9b866976d0cc05fb056ee36da8681b98",
            "type": "DEPLOY"
        }
    ]
}
```

**Expected behavior**
Block number 1 has transaction_hash ```0x567ec53cc336686b56ee6d466dad00b9b866976d0cc05fb056ee36da8681b98``` which is 64 characters but expected value would be 0x0.

**Screenshots**
None

**Devnet version**

- I am using Devnet version: 0.2.10
- [ ] This happens with a dockerized Devnet (check the box if true).
- This does not appear on the following Devnet version:

**System specifications**

Not related

**Comment by mikiw** [comment]
Closing, this will be solved in a new task #248.

**Comment by mikiw** [comment]
Closing, this will be solved in a new task #248.

**Comment by mikiw** [comment]
It turned out that ```transaction_hash``` is calculated in ```InternalTransaction```.

In the starknet_wrapper.py we have:
```
if self.config.lite_mode_deploy_hash:
   tx_hash = self.transactions.get_count()
else:
   tx_hash = internal_tx.hash_value
```

So this ```transaction_hash``` is already been calculated so this optimization currently doesn't make sense. What I'm suggesting in new task is to talk with starkware team to provide something like LiteInternalTransaction class without cryptographic function. For testing purposes, this new class will not calculate hashes.
